contracts/test/CompoundGovernorCancelTest.t.sol

```diff
@@ -0,0 +1,158 @@
+        _randomIndex = bound(_randomIndex, 0, _majorDelegates.length - 1);
+        address _proposer = _majorDelegates[_randomIndex];
+        Proposal memory _proposal = _buildAnEmptyProposal();
+        uint256 _proposalId = governor.hashProposal(
+            _proposal.targets, _proposal.values, _proposal.calldatas, keccak256(bytes(_proposal.description))
+        );
+        _submitPassAndQueueProposal(_proposer, _proposal);
+
+        vm.prank(_proposer);
+        governor.cancel(
+            _proposal.targets, _proposal.values, _proposal.calldatas, keccak256(bytes(_proposal.description))
+        );
+        vm.assertEq(uint256(governor.state(_proposalId)), uint256(IGovernor.ProposalState.Canceled));
+    }
+
+    function testFuzz_ProposalGuardianCanCancelAnyProposal(uint256 _randomIndex) public {
+        _randomIndex = bound(_randomIndex, 0, _majorDelegates.length - 1);
+        address _proposer = _majorDelegates[_randomIndex];
+        Proposal memory _proposal = _buildAnEmptyProposal();
+        uint256 _proposalId = governor.hashProposal(
+            _proposal.targets, _proposal.values, _proposal.calldatas, keccak256(bytes(_proposal.description))
+        );
+        _submitPassAndQueueProposal(_proposer, _proposal);
+
+        vm.prank(proposalGuardian.account);
+        governor.cancel(
+            _proposal.targets, _proposal.values, _proposal.calldatas, keccak256(bytes(_proposal.description))
+        );
+        vm.assertEq(uint256(governor.state(_proposalId)), uint256(IGovernor.ProposalState.Canceled));
+    }
+
+    function testFuzz_AnyoneCanCancelAProposalBelowThreshold(address _caller) public {
+        vm.assume(_caller != COMPOUND_COMPTROLLER && _caller != PROXY_ADMIN_ADDRESS);
+        Proposal memory _proposal = _buildAnEmptyProposal();
+        uint256 _proposalId = governor.hashProposal(
+            _proposal.targets, _proposal.values, _proposal.calldatas, keccak256(bytes(_proposal.description))
+        );
+        _submitPassAndQueueProposal(delegatee, _proposal);
+        _removeDelegateeVotingWeight();
+
+        vm.prank(_caller);
+        governor.cancel(
+            _proposal.targets, _proposal.values, _proposal.calldatas, keccak256(bytes(_proposal.description))
+        );
+        vm.assertEq(uint256(governor.state(_proposalId)), uint256(IGovernor.ProposalState.Canceled));
+    }
+
+    function test_WhitelistGuardianCanCancelWhitelistedProposalBelowThreshold() public {
+        Proposal memory _proposal = _buildAnEmptyProposal();
+        uint256 _proposalId = governor.hashProposal(
+            _proposal.targets, _proposal.values, _proposal.calldatas, keccak256(bytes(_proposal.description))
+        );
+        vm.prank(whitelistGuardian);
+        governor.setWhitelistAccountExpiration(delegatee, block.timestamp + 2_000_000);
+        vm.assertTrue(governor.isWhitelisted(delegatee));
+        _submitPassAndQueueProposal(delegatee, _proposal);
+        _removeDelegateeVotingWeight();
```

Comment:
this method should take an address IMO. is it possible via some cheatcode?

Reply:
Reimplemented with the `mockCall` to take in any address and return 0 for `Comp.priorVotes` call in https://github.com/ScopeLift/compound-governance-upgrade/pull/36/commits/9bd64dff3c1163359d243bf8b601ea7b6634217b